rest-service/manager_rest/storage/resource_models.py

```diff
@@ -598,19 +598,26 @@ def _is_workflow_available(self, workflow):
     def _true_or_none(self, rule):
         return rule is True or rule is None
 
-    def _node_instances_active_states_match(self, node_instance_active_rule):
-        if node_instance_active_rule is None:
+    def _node_instances_active_states_match(self, node_instance_active_rules):
+        if node_instance_active_rules is None:
             return True
         ni_states = set(ni.state for n in self.nodes for ni in n.instances)
-        if node_instance_active_rule == 'all':
-            return all(state == 'started' for state in ni_states)
-        if node_instance_active_rule == 'some':
-            return any(state == 'started' for state in ni_states)
-        if node_instance_active_rule == 'none':
-            return not any(state == 'started' for state in ni_states)
-        raise manager_exceptions.InvalidWorkflowAvailabilityRule(
-            "Invalid value for 'node_instances_active' availability rule: "
-            f"'{node_instance_active_rule}'")
+        result = False
+        for rule in node_instance_active_rules:
+            if rule == 'all':
+                result |= all(state == 'started' for state in ni_states)
+            elif rule == 'partial':
+                result |= any(state == 'started' for state in ni_states) \
+                          and any(state != 'started' for state in ni_states)
+            elif rule == 'none':
+                result |= not any(state == 'started' for state in ni_states)
+            else:
+                raise manager_exceptions.InvalidWorkflowAvailabilityRule(
+                    "Invalid value for 'node_instances_active' availability "
+                    f"rule: '{node_instance_active_rules}'")
+            if result is True:
+                return True
+        return False
```

Comment:
hmm, since you made the result into a set, this can be written simpler
```suggestion
        result = False
        for rule in node_instance_active_rules:
            if rule == 'all':
                result |= (ni_states == {'started'})
            elif rule == 'partial':
                result |= (len(ni_states) > 1 and 'started' in ni_states)
            elif rule == 'none':
                result |= ('started' not in ni_states)
            else:
                raise manager_exceptions.InvalidWorkflowAvailabilityRule(
                    "Invalid value for 'node_instances_active' availability "
                    f"rule: '{node_instance_active_rules}'")
        return result
```
or even:
```suggestion
        for rule in node_instance_active_rules:
            if rule == 'all':
                if ni_states != {'started'}:
                    return False
            elif rule == 'partial':
                if 'started' not in ni_states or ni_states == {'started'}:
                    return False
            elif rule == 'none':
                if 'started' in ni_states:
                    return False
            else:
                raise manager_exceptions.InvalidWorkflowAvailabilityRule(
                    "Invalid value for 'node_instances_active' availability "
                    f"rule: '{node_instance_active_rules}'")
        return True
```